### PR TITLE
feat(aws-codepipeline-actions) allows custom role for code commit event role

### LINKED ISSUE #10069: [aws-codepipeline-actions] No option to custom event role on code commit source action
I want to use my custom role that is be assumed by "events.amazonaws.com" to StartPipelineExecution.
 
`aws-codepipeline-actions/lib/codecommit/source-action` class is binding to `aws-events-target/lib/codepipeline` as target on commit to repository.

`aws-events-target/lib/codepipeline` provides option to pass `eventRole` but there is no option to pass event role in source-action.

### Use Case

I have a role that i need to pass as my event role.

### Proposed Solution

Can have `eventRole` inside `ActionBindOptions` and that can be passed to target codepipeline action

```ts
      this.props.repository.onCommit(`${stage.pipeline.node.uniqueId}${branchIdDisambiguator}EventRule`, {
        target: new targets.CodePipeline(stage.pipeline, **options.eventRole**),
        branches: [this.branch],
      });
```

### Other

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by nirvana124**
With this PR you  can use custom role that is assumed by "events.amazonaws.com" to start CodePipeline execution.

Fixes #10069

----
*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by skinny85 — CHANGES_REQUESTED** [review]
Thanks for the contribution @nirvana124 !

Some suggestions below.

File: packages/@aws-cdk/aws-codepipeline-actions/README.md [hunk 1]
```diff
@@ -42,6 +42,19 @@ pipeline.addStage({
 });
 ```
 
+If you want to use existing role which can be used by on commit event rule.
+You can specify the role object in triggerRole property.
+
+```ts
+const triggerRole = new iam.Role.fromRoleArn(this, 'Trigger-role', 'roleArn');
```
[the remark this answers is not in the document]
Comment (nirvana124) [reply packages/@aws-cdk/aws-codepipeline-actions/README.md]: My bad, fixed

File: packages/@aws-cdk/aws-codepipeline-actions/lib/codecommit/source-action.ts [hunk 2]
```diff
@@ -77,6 +77,14 @@ export interface CodeCommitSourceActionProps extends codepipeline.CommonAwsActio
...
    */
   readonly repository: codecommit.IRepository;
+
+  /**
+   * Role to be used by on commit event rule.
+   * Used only when trigger value is CodeCommitTrigger.EVENTS.
+   *
+   * @default a new role will be created.
+   */
+  readonly triggerRole?: iam.IRole;
```
[the remark this answers is not in the document]
Comment (nirvana124) [reply packages/@aws-cdk/aws-codepipeline-actions/lib/codecommit/source-action.ts]: renamed to eventRole

**Review by skinny85 — APPROVED** [review]
Thanks @nirvana124 !

**Pull request merged**